dev-client/src/screens/BottomTabsScreen.tsx

```diff
@@ -38,19 +37,24 @@ import {useDispatch} from 'terraso-mobile-client/store';
 export const BottomTabsScreen = memo(() => {
   const dispatch = useDispatch();
   const {keyboardStatus} = useKeyboardStatus();
-  const [potentiallyOutdatedLocationPermission, _, requestLocationPermission] =
-    useForegroundPermissions();
+  const [locationPermissions, requestLocationPermissions, _] =
+    useUpdatedForegroundPermissions();
 
   useEffect(() => {
-    if (!potentiallyOutdatedLocationPermission?.granted) {
-      requestLocationPermission();
+    const requestAndAwaitPermissions = async () => {
+      await requestLocationPermissions();
+    };
+
+    if (!locationPermissions?.granted) {
+      requestAndAwaitPermissions();
```

Comment:
question: can you call ` await requestLocationPermissions()` directly?

Reply:
Ah thanks! 
Details: You can't call `await requestLocationPermissions()` directly in the `useEffect` because using await requires the calling function to be async and you can't make a useEffect async, so if you want to call something with `await` you need to make a separate async function and call it this way ([source](https://devtrium.com/posts/async-functions-useeffect)). 
However, I think we actually can just call `requestLocationPermissions()` here because a) the updated permissions hook should handle the permissions changing, and b) we weren't actually waiting for that to complete before starting the next useEffect in the first place.